### ISSUE TITLE
Research existing open source TTS models

### ISSUE THREAD

**Issue by iboB** [alpaca-core/ac-local#178]
https://github.com/balisujohn/tortoise.cpp is what we're about to use.

Is there anything better? Report as comment in this issue.

**Comment by pminev** [comment]
I've searched for most used TTS models and they are:
- [Ultravox](https://github.com/fixie-ai/ultravox)
  - Pros:
    - it's suitable for real-time - "time-to-first-token (TTFT) of approximately 150ms, and a tokens-per-second rate of ~60, all using a Llama 3.1 8B backbone"
    - Actively developing
  - Cons:
    - Doesn't have .cpp version
    - Doesn't support voice cloning
- [Tortoise](https://github.com/neonbjb/tortoise-tts) / [tortoise.cpp](https://github.com/balisujohn/tortoise.cpp)
  - Pros:
    - Has some of the most natural-sounding speech
    - Excels at generating multiple distinct voices
    - Can clone voices
    - Has .cpp version
  - Cons:
    - Occasionally deviates from the exact input text
    - The developer behind it was hired from OpenAI, so it might be not so actively developing
- [Bark](https://github.com/suno-ai/bark)
  - Pros:
    - Highly realistic, multilingual speech
    - Generates music, background noise, simple sound effects
  - Cons: 
    - Doesn't have .cpp version
    - Not actively developing
- [MeloTTS](https://github.com/myshell-ai/MeloTTS)
  - Pros:
    - It says it's fast enough for real-time CPU and supports multiple languages
  - Cons:
    - No support for voice cloning
    - Doesn't have .cpp version
- [StyleTTS2](https://github.com/yl4579/StyleTTS2)
  - Pros:
    -  Realistic and natural speech
  - Cons
    - Not actively developing
    - Doesn't have .cpp version
- [ChatTTS](https://github.com/2noise/ChatTTS)
   - AGPLv3+ license

**Comment by iboB** [comment]
Ok. Let's evaluate tortoise deeper since it's the only one with a .cpp version. 

Try running it locally. How does it sound? How fast is it?

**Comment by pminev** [comment]
I forked [tortoise.cpp](https://github.com/alpaca-core/tortoise.cpp/tree/feature/metal-support) in our org and in the beginning I was able to run it only on CPU because some operations were not ported for Metal (they were implemented only for CUDA). So, after a couple of days I was finally ready with implementation and bug-fixing(there were bugs in metal part in ggml since the guy was not using vanilla ggml).

The sound is good, the same as the one on the CPU. For small sentence I tested with ("dr freeman, does the metal work?") the results are:
- CPU - 133.31 sec 
- Metal - 23.21 sec (almost 6x faster)
